bgrgicak/woocommerce
# Add themes to in app search

### Changes proposed in this Pull Request:

This PR introduces themes to in-app search. To do this it moves the search request from `Extension` and `Theme` into `Content`. It adds a `SearchResults` component that splits the data and loads `Extension` and `Theme` sections.

To get and split the data this PR adds types to `Product` and the ability to fetch themes and extension in a single API call.

While debugging some flaky behavior I noticed issues with how we update the loader and initially load components and fixed it in the updated `Content` component.

The search results tab will load only while searching.



### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. [Open the extensions page without a set tab](http://localhost:8888/wp-admin/admin.php?page=wc-admin&path=%2Fextensions)
2. Confirms it loads the Discover tab
3. Open network tools to to see `/wp-json/wccom-extensions/1.0/search` API requests
4. Open the _Browse_ tab
5. See that extension cards are loading and only one search request was made
6. Open the _Themes_ tab
7. See that theme cards are loading and only one search request was made
8. [Search for a term in header search](http://localhost:8888/wp-admin/admin.php?page=wc-admin&path=%2Fextensions&tab=search&term=store&category=)
9. Confirm that both themes and extensions are loading with the correct cards
10. Confirm the _Search  results_ tab is visible while on the search page

**Note**: The search page may not fully match the design, some features are missing, and categories are the same for themes and extensions. This will be addressed in future PRs.



### Changelog entry




<details>

#### Significance




#### Type




#### Message

Add support for themes to in-app search.

#### Comment

</details>
